source/includes/write/transaction.kt

```diff
@@ -0,0 +1,58 @@
+import com.mongodb.*
+import com.mongodb.client.ClientSession
+import com.mongodb.client.MongoClients
+import org.bson.Document
+
+// start-data-class
+data class Restaurant(val name: String, val cuisine: String)
+// end-data-class
+
+fun main() {
+// start-transaction
+    // Creates a new MongoClient with a connection string
+    val client = MongoClients.create("<connection string>")
+
+    // Gets the database and collection
+    val database = client.getDatabase("sample_restaurants")
+    val collection = database.getCollection<Restaurant>("restaurants")
+    
+    // Defines options and inserts restaurants into the collection
+    fun insertRestaurantsInTransaction(session: ClientSession) {
+        // Sets options for the collection operation within the transaction
+        val restaurantsCollectionWithOptions = collection
+            .withWriteConcern(WriteConcern("majority"))
+            .withReadConcern(ReadConcern.LOCAL)
+
+        // Inserts restaurants within the transaction
+        restaurantsCollectionWithOptions.insertOne(
+            session,
+            Restaurant("name", "Kotlin Sync Pizza").append("cuisine", "Pizza")
```

Comment:
S: I think these Restaurant instantiations are incorrect. They should look more like:
```kotlin
Restaurant("Kotlin Sync Pizza", "Pizza")
``` 

I would recommend opening the file in a jvm IDE like intellij to get hints

Reply:
Ah you're right, forgot to update this when I switched to the data class. Ty!